allauth/socialaccount/providers/oauth2/views.py

```diff
@@ -64,12 +64,17 @@ def view(request, *args, **kwargs):
                 return e.response
         return view
 
-    def get_client(self, request, app):
+    def get_callback_url(self, request, app):
```

Comment:
Wouldn't the OAuth2Adapter be a better place to move this into? You already have a BattleNetOAuth2Adapter ...